debezium-server-voyager/debezium-server-voyagerexporter/src/main/java/io/debezium/server/ybexporter/YbExporterConsumer.java

```diff
@@ -80,6 +93,97 @@ void connect() throws URISyntaxException {
         flusherThread = new Thread(this::flush);
         flusherThread.setDaemon(true);
         flusherThread.start();
+
+        // Register shutdown hook to release lock on exit
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            releaseLockFile();
+        }));
+    }
+
+    /**
+     * Reads the lock file and checks if the PID is still running.
+     * If the PID is not running, it deletes the lock file.
+     * If the PID is running, it throws an IllegalStateException.
+     */
+    private void readLockFileAndCheckPid() {
+        try {
+            String lockContent = Files.readString(lockFile.toPath());
+            String[] lines = lockContent.split("\n");
+            if (lines == null || lines.length == 0) {
+                // If the lock file is empty, error out
+                String msg = String.format("Lock file {} is empty.", lockFile.getAbsolutePath());
+                throw new IllegalStateException(msg);
+            }
+            String pid = lines[0].trim();
+            // Check if the PID is a valid number
+            LOGGER.info("Lock file {} exists with PID: {}", lockFile.getAbsolutePath(), pid);
+            //Parse PID and check if the process is running
+            try {
+                long pidLong = Long.parseLong(pid);
+                // Check if the process with this PID is running
+                if (ProcessHandle.of(pidLong).isPresent()) {
+                    // Process is running, error out
+                    String msg = String.format("Lock file %s already exists and process with PID %s is running. Another process may be running for this dataDir.", lockFile.getAbsolutePath(), pid);
+                    LOGGER.error(msg);
+                    throw new IllegalStateException(msg);
+                } else {
+                    // Process is not running, we can safely delete the lock file
+                    LOGGER.warn("Lock file {} exists but process with PID {} is not running. Deleting it.", lockFile.getAbsolutePath(), pid);
+                    releaseLockFile();
+                }
+            } catch (NumberFormatException e) {
+                // If PID is not a valid number, treat it as a stale lock
+                String msg = String.format("Invalid PID in lock file {}: {}. Deleting lock file.", lockFile.getAbsolutePath(), pid);
+                throw new IllegalStateException(msg, e);
+            }
+            
+        } catch (IOException e) {
+            String msg = String.format("Error reading lock file %s: %s", lockFile.getAbsolutePath(), e.getMessage());
+            LOGGER.error(msg, e);
+            throw new IllegalStateException(msg, e);
+        }
+    }
+    /**
+     * Acquires a lock file in the dataDir with <exporter-role>.lck and stores the pid of the process 
+     * Fails if already locked.
+     */
+    private void acquireLockFile() {
+        lockFile = new File(dataDir, String.format(".%s.lck", exporterRole));
```

Comment:
where do all the commands' lock files get created? Would be good to use the same directory as that.